.travis.yml

```diff
@@ -1,129 +1,78 @@
-        - PIP_INSTALL='pip install'
+        - CONDA_DEPENDENCIES='matplotlib=1.4 scipy'
+        - PIP_DEPENDENCIES='spectral-cube lmfit lineid-plot'
+
     matrix:
+        # Make sure that egg_info works without dependencies
         - SETUP_CMD='egg_info'
+        # Try all python versions with the latest numpy
+        - SETUP_CMD='test'
 
 matrix:
     include:
 
-        # Do a coverage test in Python 2. 
-        #- python: 2.7
-        #  env: SETUP_CMD='test --coverage'
-
         # Check for sphinx doc build warnings - we do this first because it
         # may run for a long time
         - python: 2.7
-          #env: SETUP_CMD='build_sphinx -w'  # 
           env: SETUP_CMD='build_sphinx' # don't crash for warnings
 
         # Try Astropy development version
         - python: 2.7
           env: ASTROPY_VERSION=development SETUP_CMD='test'
-        #- python: 3.3
-        #  env: ASTROPY_VERSION=development SETUP_CMD='test'
+        - python: 3.5
+          env: ASTROPY_VERSION=development SETUP_CMD='test'
 
-        # Try all python versions with the latest numpy
-        #- python: 2.6
-        #  env: SETUP_CMD='test'
+        # Try with optional dependencies disabled
         - python: 2.7
           env: SETUP_CMD='test'
-        #- python: 3.2
-        #  env: SETUP_CMD='test'
-        #- python: 3.3
-        #  env: SETUP_CMD='test'
-        - python: 3.4
-          env: SETUP_CMD='test'
+               CONDA_DEPENDENCIES='matplotlib=1.4'
+               PIP_DEPENDENCIES=''
         - python: 3.5
           env: SETUP_CMD='test'
-
+               CONDA_DEPENDENCIES='matplotlib=1.4'
+               PIP_DEPENDENCIES=''
         # Try older numpy versions
-        #- python: 3.2
-        #  env: NUMPY_VERSION=1.6 SETUP_CMD='test'
         - python: 2.7
-          env: NUMPY_VERSION=1.9 SETUP_CMD='test'
+          env: NUMPY_VERSION=1.7 SETUP_CMD='test'
         - python: 2.7
           env: NUMPY_VERSION=1.8 SETUP_CMD='test'
-        # numpy 1.5 stopped working - I think it's a conda issue
-
-before_install:
-
-
-    # Use utf8 encoding. Should be default, but this is insurance against
-    # future changes
-    - export PYTHONIOENCODING=UTF8
-
-    # http://conda.pydata.org/docs/travis.html#the-travis-yml-file
-    - wget https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh;
-    - bash miniconda.sh -b -p $HOME/miniconda
-    - export PATH="$HOME/miniconda/bin:$PATH"
-    - hash -r
-    - conda config --set always_yes yes --set changeps1 no
-    - conda update -q conda
-    - conda info -a
-
-    # UPDATE APT-GET LISTINGS
-    - sudo apt-get update
-
-    # DOCUMENTATION DEPENDENCIES
-    - if [[ $SETUP_CMD == build_sphinx* ]]; then sudo apt-get install graphviz texlive-latex-extra dvipng; fi
-    # http://about.travis-ci.org/docs/user/gui-and-headless-browsers/
-    - "export DISPLAY=:99.0"
-    - "sh -e /etc/init.d/xvfb start"
```

Comment:
@bsipocz the tests will fail unless an X-system is enabled, or at least that has been true in the past